incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -221,41 +211,47 @@ func (r *HierarchyConfigReconciler) markExisting(log logr.Logger, ns *forest.Nam
 	}
 }
 
-// syncOwner propagates the required child value from the forest to the spec if possible
-// (the spec itself will be synced next), or removes the owner value if there's a problem
-// and notifies the would-be parent namespace.
-func (r *HierarchyConfigReconciler) syncOwner(log logr.Logger, inst *api.HierarchyConfiguration, ns *forest.Namespace) {
-	if ns.Owner == "" {
+// syncOwner propagates the owner value from the annotation to the forest. If
+// the owner changed, the HNS instances will be enqueued to update state. If
+// there's a conflict with the owner and the parent, the owner namespace will be
+// enqueued to update the "SubnamespaceConflict" condition.
+func (r *HierarchyConfigReconciler) syncOwner(log logr.Logger, inst *api.HierarchyConfiguration, nsInst *corev1.Namespace, ns *forest.Namespace) {
+	oldOwner := ns.Owner
```

Comment:
Now that we're syncing these together, do we still need ns.Owner as a separate field from ns.Parent()? Maybe just a boolean `ns.IsOwned` would be good enough, and it just means "we set ns.Parent() based on the annotation in the namespace?"

Reply:
I think ns.Owner and ns.Parent are the forest version of nsInst's owner annotation and hcInst.Spec.Parent. What if the webhook is bypassed and the parent is different from the owner annotation? CMIIW

Reply:
As discussed offline, we will use owner annotation as source of truth. The parent will always match the owner annotation.

Reply:
I kept `ns.Owner` for now.

Reply:
sgtm but can you explain why?

Reply:
Ah I will remove it in a next PR since it's used a lot in `hierarchy_config.go` and `hierarchy.go` and I don't want to make this PR more complex (mostly for me to test). I will update #552 and put a todo comment in the forest.

Reply:
sg!